evap/staff/tests/test_importers.py

```diff
@@ -374,6 +387,117 @@ def test_replace_consecutive_and_trailing_spaces(self):
         success_messages, __, __ = EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=True)
         self.assertIn("The import run will create 1 courses/evaluations and 3 users", "".join(success_messages))
 
+    def test_existing_course_is_not_recreated(self):
+        self.create_existing_course()
+        excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_existing_course)
+
+        self.assertEqual(
+            len(
+                Course.objects.filter(
+                    semester=self.semester, name_de=self.existing_course.name_de, name_en=self.existing_course.name_en
+                )
+            ),
+            1,
+        )
+        __, warnings, __ = EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=False)
+        self.assertIn(
+            "Course Existing Course (Existierender Kurs) already exists with identical attributes. Course is not created and users are put into the evaluation of that course.",
+            "".join(warnings[ImporterWarning.DUPL]),
+        )
+        self.assertEqual(
+            len(
+                Course.objects.filter(
+                    semester=self.semester, name_de=self.existing_course.name_de, name_en=self.existing_course.name_en
+                )
+            ),
+            1,
+        )
+
+    def test_existing_course_degree_is_added(self):
+        self.create_existing_course()
+        excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_existing_course)
```

Comment:
So, creating the correct `memory_excel_file` may be different for each test, and yes, in this case it is the same for each of the new tests. Imo creating the correct `memory_excel_file` directly inside the test makes it better understandable. Moving all different excel data out of the methods into class variables sounds like bloating up the test class pretty much. Just my initial throughts. I dont know of any superior reasons for this. :D (I hope this was even part of your regards)
Regarding the `self.create_existing_course()`: yes seems valid to more it into `setUpTestData`.

Reply:
Hm I am not sure about moving the `self.create_existing_course()` into `setUpTestData` because that breaks the evaluation counts for other tests. We could make those counts independent of the evaluations in the database or not move `self.create_existing_course()`. What so you think?

Reply:
Adapting the existing course to be able to use the `test_enrollment_data_filedata` is a good idea.
Regarding the excel content: So I am going to put
`excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_fildata)`
into `setUpTestData` also for the already existing tests.

Reply:
Hm, this really shouldn't happen! Optimally, we would find out why this happens and do something in `setupTestData` to restore the old behavior, otherwise, let's just leave it with the call in every test and open an investigative-follow-up-issue. Sorry, you are really stumbling upon weird stuff in this PR.

Reply:
I think restoring the old behavior would either require removing the course again after creating it (which obviously does not help) or using an existing_course that is not in `default_excel_content` which in turn requires a different `excel_content` for all the new tests.
Cosidering that and what @He3lixxx said in https://github.com/e-valuation/EvaP/pull/1683#discussion_r806138012 I'll revert https://github.com/e-valuation/EvaP/commit/e789a8370541c215785c9183b41adfa55721df01. Shall I open an investigation issue after this pr is merged or is it resolved with the argument in https://github.com/e-valuation/EvaP/commit/e789a8370541c215785c9183b41adfa55721df01?

Reply:
it looks good as it is now :)

Feel free to open an issue, even if we figure it out instantly, it's nice to have it documented

Reply:
> Hm, this really shouldn't happen! Optimally, we would find out why this happens and do something in `setupTestData` to restore the old behavior, otherwise, let's just leave it with the call in every test and open an investigative-follow-up-issue. Sorry, you are really stumbling upon weird stuff in this PR.

> Feel free to open an issue, even if we figure it out instantly, it's nice to have it documented

Looking into it again, really nothing unexpected happened, it was just the combination of the two things that confused us:
- `test_enrollment_data_filedata`, the "import sheet" used for all other tests, was used for the new merge tests, thus the setup for the new tests had to create database entries for the courses that would be imported for that sheet.
- `create_exisiting_course` was moved to `setUpTestData`. For the reasons above, this method sets up the "course already exists" situation with the usual import data that normally shouldn't cause any errors -- which obviously then affects all tests, as we now basically start every test with a "course already exists" situation.

In detail:
- Warnings were unequal to {} because obviously, with one course from the import already existing, there would be a warning that the course already exists.
- Users were reduced from 23 to 22 because a user's email (123@institution.example.com) was re-used for a responsible (probably on purpose to trigger a warning / error)
- All other increments / decrements are explainable similarly.

Resolving this now: The current code is what we want (reuse error-free import data, create error/warning situation in tests that want it).